python/katana/bug/environment.py

```diff
@@ -0,0 +1,195 @@
+import datetime
+import os
+import re
+import socket
+import subprocess
+import sys
+import tempfile
+import zipfile
+from pathlib import Path
+from typing import Any, Iterable, Optional, Union
+
+_environment_capture_routines = []
+
+
+def register_capture_routine(f):
+    _environment_capture_routines.append(f)
+    return f
+
+
+@register_capture_routine
+def _capture_system(zipout: zipfile.ZipFile):
+    capture_string(
+        zipout,
+        "os.txt",
+        f"""
+            host: {socket.gethostname()}
+            uname: {os.uname()}
+            date: {datetime.datetime.now()}
+
+            {capture_command('lsb_release', '-a')}
+            """,
+    )
+
+    import katana  # Imported here to avoid cyclical import
```

Comment:
still not excited about import (or #include) in the middle

Reply:
I'll put it at the top of the function. That is better. But it cannot be at the top of the file due to dependency cycles. I'll check for an easy way to remove the cycle as well.